### ISSUE TITLE
[BUG] config tries close nil channel

### ISSUE THREAD

**Issue by jankremlacek** [micro/go-micro#1369]
hello guys ..

as I've updated to v2.3.0 config panicking (@ github.com/micro/go-micro/v2@v2.3.0/config/default.go:168) 

as before loaded config automatically initiated runtime scan, usually, I closed it by the code (config.Close()) .. since v2.3.0 it tries to close somehow itself, but on nil channel, see the line 168. even if I do not use the `config.Close()` anymore ..

**How to reproduce the bug:**
```go
import (
    "github.com/micro/go-micro/v2/config"
    "github.com/micro/go-micro/v2/config/source/file"
)

func abc() {
    conf := config.New()
    conf.Load(file.NewSource(file.WithPath("existing valid .json file")))
}
```

once the func ends, the panic is raised.

**Environment:**
Go Version: go1.14 darwin/amd64
```
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/jan/Library/Caches/go-build"
GOENV="/Users/jan/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/jan/.gvm/pkgsets/go1.14/global"
GOPRIVATE=""
GOPROXY="direct"
GOROOT="/Users/jan/.gvm/gos/go1.14"
GOSUMDB="off"
GOTMPDIR=""
GOTOOLDIR="/Users/jan/.gvm/gos/go1.14/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/jan/Code/kikot/tendua/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/q9/t0l3vqb964z3d7bm5v8hhms00000gn/T/go-build116572053=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

Thank you in advance for looking at it.

Last, but not least .. I really like what you do there .. what I miss is more in-depth documentation, as well as in-between versions, changes documentation. As I do miss it, I'm willing to help you with that. I would love to see once such an awesome platform with such a great website and documentation as for example PHP Laravel has.

**Comment by vtolstov** [comment]
if you can write the docs - its fine.
p.s. this error will be fixed by #1374

**Comment by jankremlacek** [comment]
thank you @vtolstov 

Regarding the doc, on one hand, I do not know now that deeply your codebase, on the other, I think it's a quite huge project to make the doc right; just saying. Let's not talk about it here. I just signed into `micro-services` Slack workspace, so you can find me there.
